### PR TITLE
Add nodejs docs

### PR DESCRIPTION
**Pull request by bergundy**
## What was changed:
Added nodejs docs

## Why?
It's an SDK, it needs documentation

## Checklist

1. How was this tested:
Ran everything locally.

2. Any docs updates needed?
Updated README with instructions for generating the node API reference.

### CODE REVIEW

**Comment by bergundy** [comment]
@flossypurse see the README instructions on how to generate the node docs.
It's not easy because the node SDK dev env is quite complex :/
You will need to generate the docs from the `more-documentation` branch.

**Comment by bergundy** [comment]
@flossypurse 
Rendered the NodeJS docs and included as a submodule in order to simplify builds.
Still missing 2 things:
1. Sidebar disappears when viewing nodejs reference docs
2. Netlify build failing probably due to missing submodule

**Comment by flossypurse** [comment]
> 2. Netlify build failing probably due to missing submodule
Ya, here is the error:

```
6:54:32 AM: Error checking out submodules: Submodule 'docs/node/reference' (git@github.com:temporalio/sdk-node.git) registered for path 'docs/node/reference'
Cloning into '/opt/build/repo/docs/node/reference'...
Host key verification failed.

fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:temporalio/sdk-node.git' into submodule path '/opt/build/repo/docs/node/reference' failed
Failed to clone 'docs/node/reference'. Retry scheduled
```

**Comment by bergundy** [comment]
@flossypurse perhaps using an HTTP URL will fix the netlify build.

**Comment by bergundy** [comment]
Depends on https://github.com/temporalio/snipsync/pull/27

**Comment by flossypurse** [comment]
> Depends on [temporalio/snipsync#27](https://github.com/temporalio/snipsync/pull/27)

Stamped.

File: README.md [hunk 1]
```diff
@@ -57,6 +57,22 @@ yarn install
...
+
+When a new NodeJS SDK version is published, the docs are automatically generated in the `docs` branch which the submodule points to, to update the docs pull the latest `docs` commit.
+
+The generated NodeJS docs include snippets which require snipsync, make sure snipsync runs after updating the submodule.
+
+For live editing of the NodeJS docs, clone the [sdk-node](https://github.com/temporalio/sdk-node) repository and run the following:
+
+1. Install the requirements as shown [here](https://github.com/temporalio/sdk-node#development)
+
+1. Genrerate typescript API reference with `npm run docs -- /path/to/documentation/docs/node/reference`
```
Comment (flossypurse) [inline README.md:74]:
```suggestion
1. Generate typescript API reference with `npm run docs -- /path/to/documentation/docs/node/reference`
```

File: README.md [hunk 2]
```diff
@@ -57,6 +57,22 @@ yarn install
 
 Read the [yarn CLI docs](https://classic.yarnpkg.com/en/docs/cli/) if you want to learn what yarn is doing.
 
+### Generate the NodeJS API reference
```
Comment (flossypurse) [inline README.md:60]: Seems like this should be in the README of the Node SDK itself?

File: docs/node/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Getting started
```
Comment (flossypurse) [inline docs/node/getting-started.md:1]: I want to see this converted to the introduction.md file with a title of "Node SDK introduction"

File: docs/node/hello-world.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
+# Hello world
```
Comment (flossypurse) [inline docs/node/hello-world.md:1]: Let's create a "Tutorials" category and make this comparable to the other "Hello World from scratch" tutorials in the other SDKs.

File: docs/node/logging.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Logging
```
Comment (flossypurse) [inline docs/node/logging.md:1]: Cool.

File: docs/node/meta-package.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+# temporalio
```
Comment (flossypurse) [inline docs/node/meta-package.md:1]: This page/info is a little island right now - Seems like it should be in the introduction or something? I am not a fan of having this as a stand alone page.

File: docs/node/package-initializer.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
+# @temporalio/create
```
Comment (flossypurse) [inline docs/node/package-initializer.md:1]: Where does this fit into the overall story? Shouldn't this be apart of introduction?

Seems like there is a lot of "set up" for this SDK. Right now, i don't see a clear story here -

File: docs/node/sdk-structure.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
+# SDK Structure
```
Comment (flossypurse) [inline docs/node/sdk-structure.md:1]: Again, this seems useful in a way - but as a stand alone page with out context I am not a fan.
We need to build a story for folks to follow here.

File: docs/node/workflow-scopes-and-cancellation.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
+# Scopes and Cancellation
+
+Temporal workflows have different types that can be cancelled:
```
Comment (flossypurse) [inline docs/node/workflow-scopes-and-cancellation.md:3]: Workflows, Activities, Workers, Task Queues, Signals, Queries should all be capitalized in documentation, code comments, and readable output.

File: package.json [hunk 10]
```diff
@@ -14,14 +14,14 @@
     "write-heading-ids": "docusaurus write-heading-ids"
   },
   "dependencies": {
-    "@docusaurus/core": "2.0.0-alpha.72",
-    "@docusaurus/preset-classic": "2.0.0-alpha.72",
+    "@docusaurus/core": "2.0.0-alpha.73",
```
Comment (flossypurse) [inline package.json:17]: Let's do a separate PR to upgrade both Docusaurus and Snipsync -

File: sidebars.js [hunk 11]
```diff
@@ -273,6 +252,32 @@ module.exports = {
...
+      id: 'node/reference/README',
+      label: 'Overview',
+    },
+    'node/getting-started',
+    'node/hello-world',
+    {
+      type: 'category',
+      label: 'API Reference',
+      items: [
+        { type: 'doc', id: 'node/meta-package', label: 'temporalio' },
```
Comment (flossypurse) [inline sidebars.js:267]: This won't be necessary if we don't use a submodule I think -

**Review by flossypurse — CHANGES_REQUESTED** [review]

File: docs/node/meta-package.md (hunk 6 above)
Comment (bergundy) [reply docs/node/meta-package.md:1]: Hmm.. I didn't really not a lot to say, but this needs to be documented and it fits the structure.

File: docs/node/package-initializer.md (hunk 7 above)
Comment (bergundy) [reply docs/node/package-initializer.md:1]: There's a lot of "set up", yes.
This is page gives more information about this package in the API reference context.
It could probably be restructured to surface some of the important information.

File: sidebars.js (hunk 11 above)
Comment (bergundy) [reply sidebars.js:267]: We still will need this.

**Review by bergundy — COMMENTED** [review]
Thanks @flossypurse

**Review by flossypurse — APPROVED** [review]
Def not in an ideal state, but as long as we keep a notice that it is all a work in progress, then I am OK with merging this as is and iterating from there.
